### PR TITLE
LPS-133430 | master

### PR DESCRIPTION
**Pull request by ThalitaCeli**

### CODE REVIEW

**Comment by Tim-Cao** [comment]
Just start reviewing

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/displaypagetemplate/DisplayPageTemplates.testcase [hunk 1]
```diff
@@ -77,6 +77,43 @@ definition {
 		}
 	}
 
+	@description = "This is a test for LPS-129360. Change the Portlet Friendly URLs in Display Page Templates to list display style."
+	@priority = "4"
+	test ChangeToListDisplayStyle {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/displaypagetemplate/DisplayPageTemplates.testcase:82]: ChangeDisplayStyleOfDocumentsAndMediaInDisplayPageTemplate

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/displaypagetemplate/DisplayPageTemplates.testcase [hunk 2]
```diff
@@ -77,6 +77,43 @@ definition {
 		}
 	}
 
+	@description = "This is a test for LPS-129360. Change the Portlet Friendly URLs in Display Page Templates to list display style."
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/displaypagetemplate/DisplayPageTemplates.testcase:80]: The user could change the display style of Documents and Media in display page template.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/displaypagetemplate/DisplayPageTemplates.testcase [hunk 3]
```diff
@@ -77,6 +77,43 @@ definition {
...
+	@priority = "4"
+	test ChangeToListDisplayStyle {
+		task ("Add an documents and media widget to the display page template") {
+			PageEditor.addWidget(portletName = "Documents and Media");
+
+			PageEditor.clickPublish();
+		}
+
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/displaypagetemplate/DisplayPageTemplates.testcase:88]: Add step to add a document entry
For example
```
		JSONDocument.addFileWithUploadedFile(
			dmDocumentTitle = "DM Document JPG",
			groupName = "Test Site Name",
			mimeType = "image/jpeg",
			sourceFileName = "Document_4.jpg");
```

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/displaypagetemplate/DisplayPageTemplates.testcase [hunk 4]
```diff
@@ -77,6 +77,43 @@ definition {
...
+		task ("View Content of the Web Content") {
+			WebContentNavigator.openWebContentAdmin(siteURLKey = "test-site-name");
+
+			WebContentNavigator.gotoEntryMenuItem(
+				entry = "Webcontent Title",
+				menuItem = "View Content");
+		}
+
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/displaypagetemplate/DisplayPageTemplates.testcase:107]: Add step to view document entry displayed as Cards style

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/displaypagetemplate/DisplayPageTemplates.testcase [hunk 5]
```diff
@@ -77,6 +77,43 @@ definition {
...
+		task ("Change to list display style") {
+			LexiconEntry.changeDisplayStyle(displayStyle = "list");
+
+			AssertElementPresent(
+				key_displayStyle = "list",
+				locator1 = "Icon#CURRENT_DISPLAY_STYLE_ACTIVE");
+		}
+	}
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/displaypagetemplate/DisplayPageTemplates.testcase:115]: Add step to view document entry displayed as List style

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/displaypagetemplate/DisplayPageTemplates.testcase [hunk 6]
```diff
@@ -77,6 +77,43 @@ definition {
...
+		}
+
+		task ("Add a web content") {
+			JSONWebcontent.addWebContent(
+				content = "Webcontent Content",
+				groupName = "Test Site Name",
+				title = "Webcontent Title");
+		}
+
+		task ("View Content of the Web Content") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/displaypagetemplate/DisplayPageTemplates.testcase:100]: Navigate to display page template at view mode

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/displaypagetemplate/DisplayPageTemplates.testcase [hunk 7]
```diff
@@ -77,6 +77,59 @@ definition {
...
+		}
+
+		task ("Add a document entry") {
+			JSONDocument.addFileWithUploadedFile(
+				dmDocumentTitle = "DM Document JPG",
+				groupName = "Test Site Name",
+				mimeType = "image/jpeg",
+				sourceFileName = "Document_4.jpg");
+
+			PageEditor.clickPublish();
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/displaypagetemplate/DisplayPageTemplates.testcase:94]: Move this step into 83-85 task block

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/displaypagetemplate/DisplayPageTemplates.testcase [hunk 8]
```diff
@@ -77,6 +77,59 @@ definition {
...
+			WebContentNavigator.gotoEntryMenuItem(
+				entry = "Webcontent Title",
+				menuItem = "View Content");
+		}
+
+		task ("View document entry displayed as Cards style") {
+			AssertElementPresent(
+				key_displayStyle = "cards",
+				locator1 = "Icon#CURRENT_DISPLAY_STYLE_ACTIVE");
+		}
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/displaypagetemplate/DisplayPageTemplates.testcase:120]: Add step to view document entry displayed as Cards style
For example,
```
		DMDocument.viewCardTitle(dmDocumentTitle = "${dmDocumentTitle}");
```

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/displaypagetemplate/DisplayPageTemplates.testcase [hunk 9]
```diff
@@ -77,6 +77,59 @@ definition {
...
+
+		task ("Change to List display style") {
+			LexiconEntry.changeDisplayStyle(displayStyle = "list");
+		}
+
+		task ("View document entry displayed as List style") {
+			AssertElementPresent(
+				key_displayStyle = "list",
+				locator1 = "Icon#CURRENT_DISPLAY_STYLE_ACTIVE");
+		}
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/displaypagetemplate/DisplayPageTemplates.testcase:130]: Add step to view document entry displayed as List style
The same as above LexiconList.viewListEntryTitle

**Comment by Tim-Cao** [comment]
Hi @ThalitaCeli, Could you update the case according to comments again?

**Comment by ThalitaCeli** [comment]
> Hi @ThalitaCeli, Could you update the case according to comments again?

Hi @Tim-Cao 
If you need any more changes let me know OK? 

Thank you.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/displaypagetemplate/DisplayPageTemplates.testcase [hunk 10]
```diff
@@ -77,6 +77,63 @@ definition {
...
+		task ("Change to List display style") {
+			LexiconEntry.changeDisplayStyle(displayStyle = "list");
+		}
+
+		task ("View document entry displayed as List style") {
+			LexiconList.viewListEntryTitle(listEntry = "DM Document JPG");
+
+			AssertElementPresent(
+				key_displayStyle = "list",
+				locator1 = "Icon#CURRENT_DISPLAY_STYLE_ACTIVE");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/displaypagetemplate/DisplayPageTemplates.testcase:133]: Use LexiconEntry.viewActiveDisplayStyle for readability

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/displaypagetemplate/DisplayPageTemplates.testcase [hunk 11]
```diff
@@ -77,6 +77,63 @@ definition {
...
+				entry = "Webcontent Title",
+				menuItem = "View Content");
+		}
+
+		task ("View document entry displayed as Cards style") {
+			DMDocument.viewCardTitle(dmDocumentTitle = "DM Document JPG");
+
+			AssertElementPresent(
+				key_displayStyle = "cards",
+				locator1 = "Icon#CURRENT_DISPLAY_STYLE_ACTIVE");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/displaypagetemplate/DisplayPageTemplates.testcase:121]: Use LexiconEntry.viewActiveDisplayStyle for more readable

**Comment by Tim-Cao** [comment]
Hi @ThalitaCeli, see changes https://github.com/liferay-echo/liferay-portal/compare/822ca9e4cea56a6916e9b5120972cb839d087d61...05f38f068dfb2cee76d0eeb939be9488b731af7c
https://github.com/liferay-echo/liferay-portal/pull/4975

**Comment by Tim-Cao** [comment]
ci:close

**Pull request closed without merging**